runtime/compiler/z/codegen/J9TreeEvaluator.cpp

```diff
@@ -3376,9 +3380,9 @@ J9::Z::TreeEvaluator::multianewArrayEvaluator(TR::Node * node, TR::CodeGenerator
       }
 
    TR::Register *temp2Reg = cg->allocateRegister();
-   // temp2Reg = firstDimLenReg * 16 (discontiguousArrayHeaderSizeInBytes)
+   // temp2Reg = firstDimLenReg * alignedDiscontiguousArrayHeaderSizeInBytes
    generateRRInstruction(cg, TR::InstOpCode::LGR, node, temp2Reg, firstDimLenReg);
-   generateRILInstruction(cg, TR::InstOpCode::MSGFI, node, temp2Reg, zeroArraySize);
+   generateRILInstruction(cg, TR::InstOpCode::MSGFI, node, temp2Reg, zeroArraySizeAligned);
 
    cursor = generateRRInstruction(cg, TR::InstOpCode::AGR, node, temp2Reg, temp1Reg);
    iComment("Calculates (firstDimLen * zeroArraySize) + (arrayStrideInBytes + arrayHeaderSize)");
```

Comment:
Can we replace the `zeroArraySize` here as well?

Reply:
Yes - thanks for catching that!

Reply:
Revised in commit e98874bf5b1be4188e76f129a472c4a0434cd525